atd-cr3-api/server.py

```diff
@@ -452,6 +453,7 @@ def associate_location():
             HASURA_EVENTS_SQS_URL[0:48]  # This is the length of the url with the account number
             + "/atd-vz-data-events-"     # We're going to add a prefix pattern for our ATD VisionZero queues
             + incoming_event_name        # And append the name of the incoming event
+            + environment_for_queue_name # Append the environment name to target the correct queue
```

Comment:
Previously, `incoming_event_name` contained the name of the event and environment set in the Hasura console/metadata. This is where the crossed environments happened when production metadata was applied to the staging Hasura cluster. It now comes from the CR3 API's environment variables set in `ZAPPA_SETTINGS` (`STAGING` or `PRODUCTION`)

Reply:
@mddilley i noticed we return a `403` if the event name is empty (`""`)—do we have a discrete list of event names we could validate against rather than accepting any string?

Reply:
@johnclary - that is a great idea. We could add that to the code here for sure. There are three valid events at the moment.